### ISSUE TITLE
ENH: When computing bootstrap CIs with BCa, allow estimating acceleration with random subsets of jackknife rather than full jackknife

### ISSUE THREAD

**Issue by siegeln** [scipy/scipy#21882]
### Is your feature request related to a problem? Please describe.

I'm running a bootstrap test with ~2 million data points. The methods `basic` and `percentile` work, but `BCa` requires estimating acceleration with a full jackknife resample. I can set the `batch` parameter to avoid memory errors from allocating an N^2 array, but this still requires computing `statistic` N times on N-1 points, which is O(N^2) if `statistic` is O(N).

But do we need to compute the full jackknife estimate for acceleration? Could we instead take a random subset of points to leave out, and estimate acceleration based on these? This would allow computational complexity to scale with O(sample_size * n_resamples), similarly to the other bootstrap methods.

Thanks!

### Describe the solution you'd like.

Add a parameter to `bootstrap`, e.g. `n_jackknife samples`, controlling the number of leave-one-out samples used to estimate the acceleration parameter.

### Describe alternatives you've considered.

Use "basic" or "percentile" methods for bootstrap estimation with large samples. However, these are less robust to distributional skewness.

### Additional context (e.g. screenshots, GIFs)

IIUC I think it would involve replacing these lines to instead randomly select indices: https://github.com/scipy/scipy/blob/3946a431c09c791e64283447de8da14060d42890/scipy/stats/_resampling.py#L55-L58

This could also naturally address issues like [gh-14645](https://github.com/scipy/scipy/issues/14645).

**Comment by siegeln** [comment]
I've taken a look but unfortunately haven't been able to find anything yet. To document my attempts so far:

An Introduction to the Bootstrap (Efron 1993) says "There are various ways to calculate the acceleration. The easiest to explain is given in terms of the jackknife values of a statistic…" (p. 186). But I wasn't able to find a discussion of the other ways.

[Efron (1987)](https://www.jstor.org/stable/2289144) motivates the formula based on the empirical influence function (7.2), which they use to derive the jackknife approximation (7.3) which appears in the textbook above. Haven't yet found other approximations here though.

[This stata user](https://www.stata.com/statalist/archive/2010-02/msg00531.html) runs into a similar problem with the computational cost of BCa on a large dataset, and similarly wonders whether we can take a random sample to estimate acceleration. But they later make the point that acceleration is a second-order correction which becomes less relevant for larger datasets, so they end going with BC, dropping the acceleration estimate.

In R's implementation, is [bcaboot](https://cran.r-project.org/web/packages/bcaboot/bcaboot.pdf) the package you're referencing, with the three methods being `bcajack`, `bcajack2`, and `bcapar`?

**Comment by mdhaber** [comment]
> In R's implementation, is bcaboot the package you're referencing,

No, actually - the `boot` package. You can look in the documentation, but I thought there was actually a little more relevant information in this [Cross Validated post](https://stats.stackexchange.com/questions/590257/using-the-bootstrap-estimates-to-compute-the-acceleration-parameter-for-bca-boot).
